bool running
float32 speed
